king_phisher/client/tabs/mail.py

```diff
@@ -936,50 +938,25 @@ def _webserver_url_key_press_event(self, widget, event):
 		self.url_thread = utilities.Thread(self._set_entry_completion_list_tsafe)
 		self.url_thread.start()
 
-	def _get_url_templates(self):
-		return self.application.rpc.graphql("""\
-		{ 
-			siteTemplates 
-				{
-					edges 
-						{ 
-							node 
-								{ 
-									created 
-									hostname 
-									path 
-									metadata
-										{
-											authors
-											classifiers 
-											description 
-											pages
-										}
-								}
-						}
-				}
-		}""")
-
 	def _set_entry_completion_list_tsafe(self):
 		gui_utilities.glib_idle_add_once(self.url_list_store.clear)
-		url_information = self._get_url_templates()
+		url_information = self.application.rpc.graphql_find_file('get_site_templates.graphql')
 		if not url_information:
 			return
 		for edge in url_information['siteTemplates']['edges']:
 			for page in edge['node']['metadata']['pages']:
 				urls = self._build_urls(edge['node']['hostname'], page, edge['node']['path'])
 				if not urls:
 					continue
-				for url in urls:
-					gui_utilities.glib_idle_add_once(self.url_list_store.append, [url])
-		self.logger.debug('updated webserver url autocomplete liststore')
+				gui_utilities.glib_idle_add_store_extend(self.url_list_store, urls)
+		self.logger.debug('successfully updated webserver url autocomplete liststore')
 		self.url_completion_last_load_time = time.time()
 
 	def _build_urls(self, hostname, page, path):
 		urls = []
 		if not hostname:
 			for address in self.config['server_config']['server.addresses']:
-				if address['host'] != '0.0.0.0':
+				if address['host'] != '0.0.0.0' and address['host'] != '::' and not ipaddress.ip_address(address['host'].is_link_local):
```

Comment:
Sorry, but I was thinking about this some more and I think my original suggestion wasn't a good one. So for the case of `::` that's not the only representation of the same IPv6 address. I'm thinking normalizing to an IP address either way and doing `ip.is_global or ip.is_private and not ip.is_unspecified`. That way we'd complete global and private addresses, but not unspecified ones like `0.0.0.0` or other odd things I can't think of.